pkg/cliplugins/base/options.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package base
+
+import (
+	"github.com/spf13/cobra"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type Options struct {
+	KubectlOverrides *clientcmd.ConfigOverrides
+
+	genericclioptions.IOStreams
+}
+
+// NewOptions provides an instance of Options with default values
+func NewOptions(streams genericclioptions.IOStreams) *Options {
+	return &Options{
+		KubectlOverrides: &clientcmd.ConfigOverrides{},
+		IOStreams:        streams,
+	}
+}
+
+// BindFlags binds the arguments common to all sub-commands,
+// to the corresponding main command flags
+func (o *Options) BindFlags(cmd *cobra.Command) {
+	// We add only a subset of kubeconfig-related flags to the plugin.
+	// All those with with LongName == "" will be ignored.
```

Comment:
Is this part of the exposed contract for the binder? Or is this a hack we will come to regret when that changes?